### DISCUSSION TITLE
Vim like key-bindings to change focus between widgets ?

### DISCUSSION THREAD

**Discussion by JackTheMico** [Textualize/textual#548] [Ideas]
Since users can create complicated layout with GridView, It would be convinent if there are key-bindings like Vim to use "h,j,k,l" to change focus on widgets.

For example, I want to create a email TUI, I would put some email accounts in top bar and some mailboxes of each account in sidebar. 
It would be great if I can change focus between widgets by pressing "Tab" or "shift+Tab" or "ctrl+h,j,k,l", 
and then use "h,j,k,l" in each widget to select different mailboxes or accounts or emails.

![1653663403](https://user-images.githubusercontent.com/19216598/170725180-817565cf-a000-42e7-9b29-fa9cb0d5f2bc.png)

My guess is this would be a good idea because there are bunch of people like using vim or emacs and that's including myself😃.
So what do you think?

**Comment by paulie-g** [comment]
Not involved with dev, but it would make sense to have an abstract tab order resolver that a given application could implement rather than hard-coding a particular set of behaviours + keybinds.

**Comment by michelececcacci** [comment]
I would be really interested in this.
